exchanges/gateio/gateio.go

```diff
@@ -448,145 +1041,2947 @@ func (g *Gateio) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange.U
+		params.Set("side", side)
+	}
+	params.Set("contract", contract.String())
+	var response []Order
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot,
+		http.MethodDelete,
+		fmt.Sprintf(futuresOrders, settle), params, nil, &response)
+}
+
+// PlaceBatchFuturesOrders creates a list of futures orders
+// Up to 10 orders per request
+// If any of the order's parameters are missing or in the wrong format, all of them will not be executed, and a http status 400 error will be returned directly
+// If the parameters are checked and passed, all are executed. Even if there is a business logic error in the middle (such as insufficient funds), it will not affect other execution orders
+// The returned result is in array format, and the order corresponds to the orders in the request body
+// In the returned result, the succeeded field of type bool indicates whether the execution was successful or not
+// If the execution is successful, the normal order content is included; if the execution fails, the label field is included to indicate the cause of the error
+// In the rate limiting, each order is counted individually
+func (g *Gateio) PlaceBatchFuturesOrders(ctx context.Context, settle string, args []OrderCreateParams) ([]Order, error) {
+	if settle != settleBTC && settle != settleUSD && settle != settleUSDT {
+		return nil, errMissingSettleCurrency
+	}
+	if len(args) > 10 {
+		return nil, errTooManyOrderRequest
+	}
+	for x := range args {
+		if args[x].Size == 0 {
+			return nil, fmt.Errorf("%v, specify positive number to make a bid, and negative number to ask", errInvalidOrderSide)
+		}
+		if !(args[x].TimeInForce == gctTIF ||
+			args[x].TimeInForce == iocTIF ||
+			args[x].TimeInForce == pocTIF ||
+			args[x].TimeInForce == focTIF) {
+			return nil, errInvalidTimeInForce
+		}
+		if args[x].Price > 0 && args[x].TimeInForce == iocTIF {
+			args[x].Price = 0
+		}
+		if args[x].Price < 0 {
+			return nil, errInvalidPrice
+		}
+		if args[x].Text != "" && !strings.HasPrefix(args[x].Text, "t-") {
+			args[x].Text = "t-" + args[x].Text
+		}
+		if args[x].AutoSize != "" && (args[x].AutoSize == "close_long" || args[x].AutoSize == "close_short") {
+			return nil, errInvalidAutoSizeValue
+		}
+		if !(args[x].Settle == settleBTC || args[x].Settle == settleUSD || args[x].Settle == settleUSDT) {
+			return nil, errMissingSettleCurrency
+		}
+	}
+	var response []Order
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot,
+		http.MethodPost,
+		fmt.Sprintf(futuresBatchOrders, settle),
+		nil, &args, &response)
+}
+
+// GetSingleFuturesOrder retrives a single order by its identifier
+func (g *Gateio) GetSingleFuturesOrder(ctx context.Context, settle, orderID string) (*Order, error) {
+	if settle != settleBTC && settle != settleUSD && settle != settleUSDT {
+		return nil, errMissingSettleCurrency
+	}
+	if orderID == "" {
+		return nil, fmt.Errorf("%v, 'order_id' cannot be empty", errInvalidOrderID)
```

Comment:
%v -> %w

Theres a few instances across this PR where errors need to be wrapped.